apps/web/src/components/forms/advanced-monitor-form.tsx

```diff
@@ -178,9 +185,10 @@ export function AdvancedMonitorForm({ defaultValues, workspaceSlug }: Props) {
             <FormItem className="sm:col-span-2 sm:col-start-1 sm:self-baseline">
               <FormLabel>Method</FormLabel>
               <Select
-                onValueChange={(value) =>
-                  field.onChange(methodsEnum.parse(value))
-                }
+                onValueChange={(value) => {
+                  field.onChange(methodsEnum.parse(value));
+                  form.resetField("body");
```

Comment:
We will need an additional config here, otherwise it will be back to `defaultValue.body`, which might be defined if we switch from "POST" back to "GET". (see [docs](https://react-hook-form.com/docs/useform/resetfield)).
```suggestion
                  form.resetField("body", { defaultValue: "" });
```

Reply:
Yep, I totally missed that case. Will push another commit.